### PR TITLE
feature-request: Added 'Workspaces IP Access Control' plugin and test cases

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

**Review by dipsubha06 — APPROVED** [review]

File: plugins/aws/workspaces/workspacesIpAccessControl.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Workspaces IP Access Control',
+    category: 'Workspaces',
+    description: 'Ensures enforced IP Access Control on Workspaces',
+    more_info: 'Checking the existence of IP Access control on Workspaces and no open IP to any Workspaces',
```
Comment (giorod3) [inline plugins/aws/workspaces/workspacesIpAccessControl.js:8]:
```suggestion
    more_info: 'Checking the existence of IP Access control on Workspaces and ensuring that no Workspaces are open',
```

File: plugins/aws/workspaces/workspacesIpAccessControl.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                var arn = 'arn:' + awsOrGov + ':workspaces:' + region + ':' + accountId + ':workspace/' + workspace.WorkspaceId;
+
+                if (!workspace.DirectoryId){
+                    helpers.addResult(results, 2, disabledString, region, arn);
+                    continue;
+                }
+
+                var workspaceDirectory = listDirectories.find(directory => directory.DirectoryId === workspace.DirectoryId);
+
+                if (workspaceDirectory.ipGroupIds) {
```
Comment (giorod3) [inline plugins/aws/workspaces/workspacesIpAccessControl.js:58]:
```suggestion
                if (workspaceDirectory  && workspaceDirectory.ipGroupIds) {
```

File: plugins/aws/workspaces/workspacesIpAccessControl.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                    let openToEverything = false;
+                    for (var workspaceIPGroup of workspaceDirectory.ipGroupIds){
+                        var ipGroup = listIPGroups.find(o => o.groupId === workspaceIPGroup);
+
+                        if (ipGroup.userRules) {
```
Comment (giorod3) [inline plugins/aws/workspaces/workspacesIpAccessControl.js:63]:
```suggestion
                        if (ipGroup && ipGroup.userRules) {
```

**Review by giorod3 — CHANGES_REQUESTED** [review]
a few comments here

**Review by giorod3 — APPROVED** [review]

**Pull request merged**